scripts/codecov-check.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e
+if command -v codecov > /dev/null 2>&1; then
+  echo The command codecov is available
+else
+  echo The command codecov is not available, installing...
+  set -x
+
+  echo Importing Codecov PGP public keys...
+  curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --import
+
+  echo Downloading codecov uploader...
+  curl -Os https://uploader.codecov.io/latest/linux/codecov
```

Comment:
This will download the [new Codecov uploader](https://docs.codecov.com/docs/codecov-uploader) which is a statically compiled binary. This reduces the risk of bad actors altering the file at its source but we still perform additional validation further down to ensure it hasn't been replaced with a different file.